Signed HEAD obliterates the build system

### When reporting a bug/issue:
- [x] Screenshot:
```
$ make  fontforge
[  4%] Built target gunicode
[ 23%] Built target gutils
[ 26%] Building C object fontforge/CMakeFiles/fontforge.dir/python.c.o
In file included from /home/nabijaczleweli/uwu/fontforge/fontforge/python.c:30:
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:41:31: warning: missing terminating '"' character [-Winvalid-pp-token]
   41 | #define FONTFORGE_GIT_VERSION "gpg: Signature made Thu 10 Nov 2022 02:54:23 CET
      |                               ^
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:42:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
   42 | gpg:                using RSA key 7D69474E84028C5CC0C44163BCFD0B018D2658F1
      | ^
      | int
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:42:4: error: expected ';' after top level declarator
   42 | gpg:                using RSA key 7D69474E84028C5CC0C44163BCFD0B018D2658F1
      |    ^
      |    ;
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:42:21: error: unknown type name 'using'
   42 | gpg:                using RSA key 7D69474E84028C5CC0C44163BCFD0B018D2658F1
      |                     ^
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:42:30: error: expected ';' after top level declarator
   42 | gpg:                using RSA key 7D69474E84028C5CC0C44163BCFD0B018D2658F1
      |                              ^
      |                              ;
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:43:27: warning: missing terminating '"' character [-Winvalid-pp-token]
   43 | gpg: Good signature from \"наб <nabijaczleweli@nabijaczleweli.xyz>\" [ultimate]
      |                           ^
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:44:27: warning: missing terminating '"' character [-Winvalid-pp-token]
   44 | gpg:                 aka \"наб <nabijaczleweli@gmail.com>\" [ultimate]
      |                           ^
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:45:41: warning: missing terminating '"' character [-Winvalid-pp-token]
   45 | 4c63c6e81e286b4210e0139798669ae45c97ccd9"
      |                                         ^
In file included from /home/nabijaczleweli/uwu/fontforge/fontforge/python.c:34:
In file included from /home/nabijaczleweli/uwu/fontforge/fontforge/autohint.h:4:
In file included from /home/nabijaczleweli/uwu/fontforge/fontforge/edgelist.h:31:
In file included from /home/nabijaczleweli/uwu/fontforge/fontforge/splinefont.h:31:
In file included from /home/nabijaczleweli/uwu/fontforge/inc/basics.h:39:
In file included from /usr/include/stdlib.h:395:
/usr/include/x86_64-linux-gnu/sys/types.h:33:9: error: unknown type name '__u_char'
   33 | typedef __u_char u_char;
      |         ^
/home/nabijaczleweli/uwu/fontforge/fontforge/python.c:19637:5: warning: 'Py_SetProgramName' is deprecated [-Wdeprecated-declarations]
 19637 |     Py_SetProgramName(saved_progname);
       |     ^
/usr/include/python3.11/pylifecycle.h:37:1: note: 'Py_SetProgramName' has been explicitly marked deprecated here
   37 | Py_DEPRECATED(3.11) PyAPI_FUNC(void) Py_SetProgramName(const wchar_t *);
      | ^
/usr/include/python3.11/pyport.h:336:54: note: expanded from macro 'Py_DEPRECATED'
  336 | #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
      |                                                      ^
/home/nabijaczleweli/uwu/fontforge/fontforge/python.c:19675:7: error: expected expression
 19675 |              FONTFORGE_GIT_VERSION );
       |              ^
/home/nabijaczleweli/uwu/fontforge/build/inc/fontforge-version-extras.h:41:31: note: expanded from macro 'FONTFORGE_GIT_VERSION'
   41 | #define FONTFORGE_GIT_VERSION "gpg: Signature made Thu 10 Nov 2022 02:54:23 CET
      |                               ^
5 warnings and 6 errors generated.
make[3]: *** [fontforge/CMakeFiles/fontforge.dir/build.make:888: fontforge/CMakeFiles/fontforge.dir/python.c.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:1231: fontforge/CMakeFiles/fontforge.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1238: fontforge/CMakeFiles/fontforge.dir/rule] Error 2
make: *** [Makefile:615: fontforge] Error 2

```

- [x] The FontForge version and the operating system you're using: 8ce3216d492d4a249ffec163c005d098df28e58c and debian
- [x] The behavior you expect to see, and the actual behavior: fontforge builds normally, and it doesn't
- [x] Steps to reproduce the behavior: sign HEAD, cmake .., make
- [x] \(optional) Possible solution/fix/workaround: pass --no-show-signature

--- a/cmake/FontForgeConfigure.cmake
+++ b/cmake/FontForgeConfigure.cmake
@@ -112,7 +112,7 @@ function(_get_git_version)
     if(Git_FOUND)
       execute_process(
         COMMAND
-          "${GIT_EXECUTABLE}" "log" "--no-show-signature" "--pretty=format:%H" "-n" "1"
+          "${GIT_EXECUTABLE}" "rev-list" "-n" "1" "HEAD"
         WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
         RESULT_VARIABLE GIT_RETVAL
         OUTPUT_VARIABLE GIT_OUTPUT

--- a/cmake/FontForgeConfigure.cmake
+++ b/cmake/FontForgeConfigure.cmake
@@ -112,7 +112,7 @@ function(_get_git_version)
     if(Git_FOUND)
       execute_process(
         COMMAND
-          "${GIT_EXECUTABLE}" "log" "--pretty=format:%H" "-n" "1"
+          "${GIT_EXECUTABLE}" "log" "--no-show-signature" "--pretty=format:%H" "-n" "1"
         WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
         RESULT_VARIABLE GIT_RETVAL
         OUTPUT_VARIABLE GIT_OUTPUT

